roles/common/README.md

```diff
@@ -1,41 +1,151 @@
-Role Name
-=========
+# Common Role
 
-The are packages that are generally expected on a server that Ubuntu doesn't
-always have on by default.
+Install all the tools our VMS have in Common
 
-Requirements
-------------
+## Overview
 
-None
+Since we include logrotate in the role, we want other roles to have access to a highly flexible, hierarchical approach to managing log rotation across your infrastructure. It allows you to define logrotate rules at multiple levels of your Ansible inventory, from global defaults to host-specific configurations.
 
-Role Variables
---------------
+## Configuration Structure
 
-None
+### Variable Hierarchy
 
-Dependencies
-------------
+Variables can be defined at multiple levels with the following precedence (highest to lowest):
 
-None
+1. **Host Variables** (`host_vars/hostname.yml`)
+2. **Group Variables** (`group_vars/group_name.yml`)
+3. **Role Variables** (`roles/role_name/vars/main.yml`)
+4. **Global Variables** (`group_vars/all.yml`)
+5. **Role Defaults** (`roles/role_name/defaults/main.yml`)
```

Comment:
This may not be accurate, since we are not actually using group vars in the way Ansible designed them to be used. For example, we do not have a file called `group_vars/figgy_production.yml` - instead we have a directory called `group_vars/figgy` and within that we have various files . . . We typically include those variables as `vars_files`, so I'm not sure how Ansible would choose among multiple values in our case.